sympy/geometry/plane.py

```diff
@@ -401,8 +401,10 @@ def intersection(self, o):
                 a = Point3D(o.arbitrary_point(t))
                 p1, n = self.p1, Point3D(self.normal_vector)
 
-                # TODO: Replace solve with solveset, when this line is tested
-                c = solve((a - p1).dot(n), t)
+                _solset = solveset((a - p1).dot(n), t)
+                c = []
+                for _i in _solset:
+                    c.append(_i)
```

Comment:
Is the solution set guaranteed to be iterable?

Reply:
I think so, since it deals with linear entities. The solution should be either `EmptySet` or `FiniteSet`.

Reply:
What if they intersect everywhere?

Reply:
We get plane and/or line as outputs.
```
>>> p=Plane(Point3D(1, 1, 1), Point3D(2, 3, 4), Point3D(2, 2, 2))
>>> l=Line(Point3D(1, 1, 1), Point3D(2, 3, 4))
>>> p.intersection(l)
[Line3D(Point3D(1, 1, 1), Point3D(2, 3, 4))]
>>> p.intersection(p)
[Plane(Point3D(1, 1, 1), (-1, 2, -1))]
>>> l.intersection(l)
[Line3D(Point3D(1, 1, 1), Point3D(2, 3, 4))]

```

Reply:
And then what happens when iterating over the result from solveset?

Reply:
In case of intersecting everywhere, `(a - p1).dot(n)` gives zero in [this](https://github.com/sympy/sympy/blob/5288eb336d12e4f3c3070bce67d75f38ab4d8376/sympy/geometry/plane.py#L405) line.

Reply:
Another case is when both the inputs are same, i.e. `self` and `o` are the same. In such a case [this condition](https://github.com/sympy/sympy/blob/5288eb336d12e4f3c3070bce67d75f38ab4d8376/sympy/geometry/plane.py#L397) is satisfied